server/drivers/yard2LCD.c

```diff
@@ -0,0 +1,647 @@
+/***************************************************************************
+ *   Copyright (C) 2009 by M. Feser                                        *
+ *   Copyright (C) 1013 by R. Geigenberger (Update for YARD2)              *
+ *                                                                         *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <time.h>
+
+#include "lcd.h"
+//#define DEBUG
+#include "report.h"
+#include "yard2LCD.h"
+#include "lcd_lib.h"
+
+/*
+//Type of characters currently stored in CGRAM. from lcd.h as info
+typedef enum {
+	standard,		// one char is used for heartbeat
+	vbar,			// vertical bars
+	hbar,			// horizontal bars
+	icons,			// standard icons
+	custom,			// custom settings
+	bignum,			// big numbers
+} CGmode;
+*/
+
+typedef enum {
+	Dmode_standard,
+ 	Dmode_graphical
+} Dmode;
+
+
+/* Driver private data */
+typedef struct driver_private_data {
+	int fd;
+	int width, height;
+	int gwidth, gheight;
+	int hspace, vspace;
+	int cellwidth, cellheight;
+	int bigcellwidth, bigcellheight;
+	char *framebuf;
+	int on_brightness;
+	int off_brightness;
+	char hw_brightness;
+	CGmode ccmode;
+	Dmode dispmode;
+	char info[255];
+	char LCDtype;
+} PrivateData;
```

Comment:
A lot of these properties (like cellwidth) aren't used at all except for shuffling defaults back and forth. Unless you are planning to actually implement this, consider omitting this. Especially get rid of it, if the protocoll of your server doesn't support it anyway.